### PR TITLE
note breaking changes in backup and restore section

### PR DESCRIPTION
**Pull request by christinaausley**
## What is the purpose of the change

Based on the conversation in https://github.com/camunda/camunda-platform-docs/issues/1609 and changes made in https://github.com/camunda/camunda-platform-docs/pull/1614, this PR adjusts existing notes to reflect breaking changes in the `next` version of docs.

## PR Checklist

- [x] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: docs/self-managed/backup-restore/backup-and-restore.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ keywords: ["backup", "backups"]
 ---
 
 :::note
-This API is subject to change.
+This release introduces breaking changes for Operate and Tasklist. See [backup and restore Operate and Tasklist data](./operate-tasklist-backup.md) for details.
```
Comment (christinaausley) [inline docs/self-managed/backup-restore/backup-and-restore.md:9]: @RomanJRW and @Zelldon I'm noting Operate and Tasklist breaking changes here, but are there any for Optimize or Zeebe in the `next` version of docs? Are these still subject to change?

File: docs/self-managed/backup-restore/operate-tasklist-backup.md [hunk 2]
```diff
@@ -5,6 +5,12 @@ description: "How to perform a backup and restore of Operate and Tasklist data."
 keywords: ["backup", "backups"]
 ---
 
+:::note
+This release introduces breaking changes, including the [get backup state API and response codes](#get-backup-state-api), [get backups list API](#get-backups-list-api), and notably the utilized URL.
+
+For example, `curl 'http://localhost:8080/actuator/backups'` rather than the previously used `backup`.
```
Comment (christinaausley) [inline docs/self-managed/backup-restore/operate-tasklist-backup.md:11]: @sdorokhova Can we make a note here on backwards compatibility or is this still subject to change?

File: docs/self-managed/backup-restore/backup-and-restore.md (hunk 1 above)
Comment (RomanJRW) [reply docs/self-managed/backup-restore/backup-and-restore.md:9]: @PHWaechtler - can you answer here please for the Optimize side

Comment (Zelldon) [reply docs/self-managed/backup-restore/backup-and-restore.md:9]: I guess @deepthidevaki knows best regarding backup

Comment (PHWaechtler) [reply docs/self-managed/backup-restore/backup-and-restore.md:9]: Yes I would say the same applies for Optimize as we've made some of the same changes as operate/tasklist, eg renaming `/backup` to `/backups`

Comment (deepthidevaki) [reply docs/self-managed/backup-restore/backup-and-restore.md:9]: @christinaausley There are similar API changes in Zeebe as well. The new API is already documented. But we didn't document it as a breaking change as it was an experimental feature in 8.1.

Comment (christinaausley) [reply docs/self-managed/backup-restore/backup-and-restore.md:9]: Thanks all!

**Review by ralfpuchert — DISMISSED** [review]

File: docs/self-managed/backup-restore/operate-tasklist-backup.md (hunk 2 above)
Comment (christinaausley) [reply docs/self-managed/backup-restore/operate-tasklist-backup.md:11]: @sdorokhova is on vacation -- I'll launch an additional PR depending on feedback upon return.

**Review by akeller — APPROVED** [review]
Thank you!

**Pull request merged**